keras_core/layers/preprocessing/discretization.py

```diff
@@ -241,20 +237,15 @@ def call(self, inputs):
         if not isinstance(
             inputs,
             (
-                tf.Tensor,
-                tf.SparseTensor,
-                tf.RaggedTensor,
                 np.ndarray,
                 backend.KerasTensor,
             ),
         ):
-            inputs = tf.convert_to_tensor(
-                backend.convert_to_numpy(inputs), dtype=self.input_dtype
+            inputs = backend.convert_to_tensor(
+                inputs, dtype=self.input_dtype
             )
 
-        from keras_core.backend.tensorflow.numpy import digitize
-
-        indices = digitize(inputs, self.bin_boundaries)
+        indices = ops.digitize(inputs, self.bin_boundaries)
 
         outputs = tf_utils.encode_categorical_inputs(
```

Comment:
This function needs to be rewritten to be backend agnostic.

Reply:
done